java/semantickernel-api/src/main/java/com/microsoft/semantickernel/contextvariables/converters/PrimitiveBooleanVariableContextVariableTypeConverter.java

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft. All rights reserved.
+package com.microsoft.semantickernel.contextvariables.converters;
+
+import com.microsoft.semantickernel.contextvariables.ContextVariableTypeConverter;
+import com.microsoft.semantickernel.contextvariables.ContextVariableTypes;
+import java.util.Locale;
+
+/**
+ * A {@link ContextVariableTypeConverter} for {@link Boolean} variables. Use
+ * {@code ContextVariableTypes.getDefaultVariableTypeForClass(Boolean.class)} to get an instance of
+ * this class.
+ *
+ * @see ContextVariableTypes#getGlobalVariableTypeForClass(Class)
+ */
+public class PrimitiveBooleanVariableContextVariableTypeConverter extends
+    PrimitiveVariableContextVariableTypeConverter {
+
+    /**
+     * Initializes a new instance of the
+     * {@link PrimitiveBooleanVariableContextVariableTypeConverter} class.
+     */
+    public PrimitiveBooleanVariableContextVariableTypeConverter() {
+        super(
+            boolean.class,
+            s -> {
+                switch (((String) s).toLowerCase(Locale.ROOT).trim()) {
```

Comment:
I wonder if telling the model to return values that have a boolean semantic as 'true' or 'false' would work. 
Would it be better to have the list in a resource that can be modified if needed? What if I want to add "dah" or "nyet"?

Reply:
I will do that in a seperate PR